packages/@aws-cdk/tmp-toolkit-helpers/src/api/refactoring/digest.ts

```diff
@@ -82,10 +89,29 @@ export function computeResourceDigests(template: CloudFormationTemplate): Record
   const result: Record<string, string> = {};
   for (const id of order) {
     const resource = resources[id];
-    const depDigests = Array.from(graph[id]).map((d) => result[d]);
-    const propsWithoutRefs = hashObject(stripReferences(stripConstructPath(resource)));
-    const toHash = resource.Type + propsWithoutRefs + depDigests.join('');
-    result[id] = crypto.createHash('sha256').update(toHash).digest('hex');
+
+    const model = loadResourceModel(resource.Type);
+    const primaryIdentifier = intersection(Object.keys(resource.Properties ?? {}), model?.primaryIdentifier ?? []);
```

Comment:
If `primaryIdentifier` is compound, it's not "one of", it's "all of".

An `intersection` would be nonempty if at least one of the keys matched, but don't we need all keys to match?